osf/management/commands/migrate_egap_registrations.py

```diff
@@ -0,0 +1,286 @@
+    :param row: Row in spreadsheet
+    :param egap_author OSFUser - OSFUser associated with all EGAP registrations
+    :param draft: DraftRegistration
+    :return registration_metdata: Dictionary
+    """
+    registration_metadata = {}
+
+    for question in schema_to_spreadsheet_mapping:
+        qid = question.keys()[0]
+        column_name = question.values()[0]
+        value = build_question_response(header_row, row, qid, column_name)
+        validated_qid, other_response = validate_response(qid, value, draft)
+        registration_metadata[validated_qid] = value
+        if other_response:
+            registration_metadata[other_response] = build_nested_response('Other (describe in text box below)')
+
+    # q35 and q36 are required questions at the end of the schema, certification and
+    # confirmation questions. Just marking as agree -
+    registration_metadata['q35'] = build_nested_response('Agree')
+    registration_metadata['q36'] = build_nested_response('Agree')
+    return registration_metadata
+
+def create_draft_registration(project, egap_author, schema):
+    """Create a draft registration with empty metadata
+    :param project Node - node to register
+    :param egap_author OSFUser - OSFUser associated with all EGAP registrations
+    :param schema RegistrationSchema - EGAP Registration Schema
+    :return DraftRegistration
+    """
+    draft_registration = DraftRegistration.create_from_node(
+        project,
+        user=egap_author,
+        schema=schema,
+        data={}
+    )
+    return draft_registration
+
+def make_projects_and_draft_registrations(schema):
+    """ Create a project and a draft registration for every row in the EGAP data dump
+    For each row in the EGAP dump:
+        1) create a project with title, egap authors, and remaining authors.
+        2) create an EGAP draft registration linked to the project, copying over all answers to supplemental questions
+    :param schema: RegistrationSchema, EGAP
+    :return:
+    """
+    with open('egap_registry_for_cos.csv') as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=',')
+        header_row = next(csv_reader)
+        normalized_header_row = [col_header.decode('ascii', 'ignore') for col_header in header_row]
+
+        egap_author = UserFactory()  # TODO, this would need to be the EGAP user (Matthew Lisiecki)
+        title_index = normalized_header_row.index('TITLE')
+        authors_index = normalized_header_row.index('B2 AUTHORS')
+
+        for line in csv_reader:
+            row = [cell.decode('ascii', 'ignore').strip() for cell in line]
+            project = create_egap_project(row[title_index], row[authors_index], egap_author)
+            draft_registration = create_draft_registration(project, egap_author, schema)
+
+            registration_metadata = build_registraton_metadata(project, normalized_header_row, row, egap_author, draft_registration)
+            draft_registration.update_metadata(registration_metadata)
```

Comment:
When https://github.com/pattisdr/osf.io/pull/30/files is merged, a call to `update_metadata` will flatten the `registration_metadata` and store it on the corresponding `registration_responses` field - so no need to worry about schema blocks changes at this point.